File: src/main/java/com/guestlogix/takehometest/model/Route.java
package com.guestlogix.takehometest.model;

public class Route {

	private Airline airline;
	private Airport origin;
	private Airport destination;

	private Route(Airline airline, Airport origin, Airport destination) {
		super();
		this.airline = airline;
		this.origin = origin;
		this.destination = destination;
	}

	public Airline getAirline() {
		return airline;
	}

	public Airport getOrigin() {
		return origin;
	}

	public Airport getDestination() {
		return destination;
	}

	public static Route create(Airline airline, Airport origin, Airport destination) {
		return new Route(airline, origin, destination);
	}

}

File: src/RouteSearch.Domain/Entities/Route.cs
namespace RouteSearch.Domain.Entities
{
    public class Route
    {
        public Airport Origin { get; private set; }
        public Airport Destination { get; private set; }
        public Airline Airline { get; private set; }

        public Route(Airport origin, Airport destination, Airline airline)
        {
            Origin = origin;
            Destination = destination;
            Airline = airline;
        }
    }
}